azurelinuxagent/common/types.py

```diff
@@ -0,0 +1,26 @@
+# Microsoft Azure Linux Agent
```

Comment:
I would recommend not adding this type now.

Seems to be a copy of _nameset in protocol util, which we will remove along with the metadata protocol.

This type is very simplistic and lacks a lot of functionality provided by enums (like assigning values to the items in the enum). Its design is odd since it is an extension of a set, etc. Also it may introduce subtle coding errors if we use it extensively because it doesn't really behave as an enum. For example

```
>>> Foo = Enum('Foo', 'one two three')
>>> Bar = Enum('Foo', 'one four five')
>>> f = Foo.one
>>> b = Bar.one
>>> f == b
False

```
vs

```
>>> Foo = AttributeEnum(('one', 'two', 'three'))
>>> Bar = AttributeEnum(('one', 'four', 'five'))
>>> f = Foo.one
>>> b = Bar.one
>>> f == b
True
```

If we really want an enum type we could consider using the backport of enum instead (https://pypi.org/project/enum34/), but we would need to pack it along with the agent.